component/src/main/java/org/ballerinalang/sap/nativeimpl/consumer/Start.java

```diff
@@ -57,53 +58,41 @@
 public class Start extends BlockingNativeCallableUnit {
 
     private static Log log = LogFactory.getLog(Start.class);
+    private static final PrintStream console = System.out;
 
     @Override
     public void execute(Context context) {
         Struct consumerStruct = BLangConnectorSPIUtil.getConnectorEndpointStruct(context);
         Map<String, Resource> sapService = (Map<String, Resource>) consumerStruct.getNativeData(SAP_RESOURCE);
         String transportName = (String) consumerStruct.getNativeData(CONSUMER_TRANSPORT_NAME);
-
+        StateChangedListener stateListener = new StateChangedListener();
+        ThrowableListener listener = new ThrowableListener(context, sapService);
         if (transportName.equalsIgnoreCase(SapConstants.SAP_BAPI_PROTOCOL_NAME)) {
             FunctionHandlerFactory factory = new FunctionHandlerFactory();
-            JCoServerFunctionHandler rfcConnectionHandler = new RFCConnectionHandler(sapService);
+            JCoServerFunctionHandler rfcConnectionHandler = new RFCConnectionHandler(sapService, context);
             factory.registerGenericHandler(rfcConnectionHandler);
             JCoServer jcoServer = (JCoServer) consumerStruct.getNativeData(CONSUMER_SERVER_CONNECTOR_NAME);
             //Configure the Jco Server
             jcoServer.setCallHandlerFactory(factory);
             jcoServer.setTIDHandler(new CustomServerTIDHandler());
-            jcoServer.addServerExceptionListener((server, connectionId, serverCtx, error) ->
-                    log.info("Exception occured on " + server.getProgramID() + " connection " + connectionId
-                            + error.toString()));
-            jcoServer.addServerExceptionListener((server, connectionId, serverCtx, error) ->
-                    log.info("Exception occured on " + server.getProgramID() + " connection " + connectionId
-                            + error.toString()));
-            jcoServer.addServerStateChangedListener((server, oldState, newState) ->
-                    log.info("Server state changed from " + oldState.toString() + " to " + newState.toString()
-                            + " on server with program id " + server.getProgramID()));
-            log.info("JCo Server start...........");
+            jcoServer.addServerErrorListener(listener);
+            jcoServer.addServerExceptionListener(listener);
+            jcoServer.addServerStateChangedListener(stateListener);
             // Start the JCo server
             jcoServer.start();
+            console.println("JCo Server started...........");
```

Comment:
Is this for BAPI?

Reply:
Yes

Reply:
Should we change to "BAPI Server Started" instead?